datafusion/sqllogictest/test_files/select.slt

```diff
@@ -224,12 +224,6 @@ select
 ----
 false true false true true false false true false true true false true true false false true
 
-# select NaNs
-query RRRR
-select 'NaN'::double a, '-NaN'::double b, 'NaN'::float c, '-NaN'::float d
-----
-NaN -NaN NaN -NaN
```

Comment:
Even simply literals also are impacted, i.e. platform dependent?

Reply:
I don't think so.
I believe `NaN` and `-NaN` are finally parsed [here](https://github.com/Alexhuszagh/rust-lexical/blob/main/lexical-parse-float/src/parse.rs#L929).
So, as long as `f32::NAN`, `f64::NAN`, `-f32::NAN` and `-f64::NAN` follows IEEE 754, simple literals should not be impacted.

Reply:
I think we should leave the test to show the behavior of parsing `-NaN`.

Reply:
@alamb 
`ToString::to_string` ignores the sign of NaNs so if the sign of NaNs should be printed in the test result, we need to do like [this](https://github.com/apache/arrow-datafusion/pull/7462/files#diff-4a5281b565761d7b0e1a53c80a44c3ef00d17855b418a39ddf4ceaef7230e46cL44).
But if we do so, the issue discussed in #7458 happens.
So, if we need to ensure a literal `"-NaN"::double` is parsed as `-NaN`, how about testing like [this](https://github.com/apache/arrow-datafusion/pull/7462/files#diff-aa021899ee5a6b4e3aa6812249dcf7e2fafcef0926ae287f6904188dffc60db2R229)?